notebooks/266-speculative-sampling/266-speculative-sampling.ipynb

```diff
@@ -23,8 +23,6 @@
    "id": "f97c435a",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #2.        !optimum-cli export openvino --model $main_model_id --weight-format fp16 $main_model_path
</pre><p>int8 or int4 will not be suitable here (required to install nncf only)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
i tried with all combinations and there are some problems. Surprisingly speculative decoding in int4/int8 cases takes more time:
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/5703039/4be46831-9168-4a5e-90db-5cf786408bef)

I tried to take a look to outputs, and found out that there is a strange German word Unterscheidung

>Text = Alan Turing was a British mathematician, computer scientist, and logician who made significant contributions to the development of computer science, artificial intelligence, and cryptography. **Unterscheidung** between Turing machines and Turing degrees. Turing machines and Turing degrees are two.......

I suspect that this difference is causing problems, draft and main start to generate different outputs and therefore draft cannot assist the main network anymore. If i generate lees tokens before, then times are ok, and speculative is indeed faster. 
**Short sentence before German words appear**
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/5703039/6107bd93-7495-406d-b738-b50b1ed8771a)

Accuracy issues with int8/int4 are visible with other prompts as well, this strange German word still appears
**Different prompt int4**
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/5703039/245037ce-5ab7-4241-9e58-f3eb01251a9c)

Let's keep it fp16 and investigate why int4/int8 fails